plugins/messagelayer/mana_plugin.go

```diff
@@ -296,6 +296,22 @@ func GetHighestManaNodes(manaType mana.Type, n uint) ([]mana.Node, time.Time, er
 	return bmv.GetHighestManaNodes(n)
 }
 
+// GetHighestManaNodesFraction returns the given percent `p` of highest type mana nodes in descending order.
```

Comment:
This does work well if the mana is following some zipf distribution. But does not work if mana is distributed equally. but we need to cover the latter case as well..